File: utilities/validation/java/src/main/java/io/polyglotutilities/validation/ValidationUtils.java
package io.polyglotutilities.validation;

import java.util.regex.Pattern;

public final class ValidationUtils {
    private static final Pattern EMAIL_PATTERN = Pattern.compile("^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$");

    private ValidationUtils() {
    }

    public static boolean isNonEmpty(String value) {
        return value != null && !value.trim().isEmpty();
    }

    public static boolean isEmail(String value) {
        return value != null && EMAIL_PATTERN.matcher(value).matches();
    }

    public static boolean isWithinRange(double value, double minimum, double maximum) {
        return Double.isFinite(value) && value >= minimum && value <= maximum;
    }
}

File: utilities/validation/kotlin/src/main/kotlin/io/polyglotutilities/validation/ValidationUtils.kt
package io.polyglotutilities.validation

object ValidationUtils {
    private val emailPattern = Regex("^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$")

    fun isNonEmpty(value: String?): Boolean =
        !value.isNullOrBlank()

    fun isEmail(value: String?): Boolean =
        value != null && emailPattern.matches(value)

    fun isWithinRange(value: Double, minimum: Double, maximum: Double): Boolean =
        value.isFinite() && value >= minimum && value <= maximum
}
